src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts

```diff
@@ -351,6 +405,8 @@ export class NotebookService extends Disposable implements INotebookService {
 			return undefined;
 		}
 		let uriString = notebookUri.toString();
+		// Format the URI string without query parameters
+		uriString = uriString.split('?')[0];
```

Comment:
Don't think it's a huge concern - but you can have a fragment (#) without a query. So something else to consider when we're working on this kind of logic.

Reply:
Yeah looking at URI standards, fragments can come after the query so I will have to go from query start to fragment start.

Reply:
What do you mean you'll "have to go from query start to fragment start"?

The URIs would usually be just the schema + authority + path part of the URI. Anything after that can probably be discounted.

I'm unsure of the long term effects of this since I don't know if VS Code has any sort of "rules" on what an editor URI can and cannot contain. But my guess is that they don't do this kind of checking so we have to be careful here to not break any scenarios that might be using the query or fragment parts of a URI.